src/tools/ts-generator/include/antares/ts-generator/thermal_interface.h

```diff
@@ -0,0 +1,50 @@
+/*
+** Copyright 2007-2024, RTE (https://www.rte-france.com)
+** See AUTHORS.txt
+** SPDX-License-Identifier: MPL-2.0
+** This file is part of Antares-Simulator,
+** Adequacy and Performance assessment for interconnected energy networks.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the Mozilla Public Licence 2.0 as published by
+** the Mozilla Foundation, either version 2 of the License, or
+** (at your option) any later version.
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** Mozilla Public Licence 2.0 for more details.
+**
+** You should have received a copy of the Mozilla Public Licence 2.0
+** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+*/
+# pragma once
+
+#include <array>
+
+#include <antares/study/parts/thermal/cluster.h>
+
+namespace Antares::TsGenerator
+{
+
+struct thermalInterface
+{
+    std::array<double, 8760> foDuration;
+    std::array<double, 8760> poDuration;
+    std::array<double, 8760> foRate;
+    std::array<double, 8760> poRate;
+    std::array<double, 8760> npoMin;
+    std::array<double, 8760> npoMax;
```

Comment:
Pros 
- We know that `foDuration` and its friends must of size 8760.

Cons
-  On my machine, `sizeof(thermalInterface) = 420512`, which is huge. With this, you **will** get stack overflows unless you're lucky.

Maybe you could wrap a `std::vector` inside a class that does not allow resizing ?

```cpp
class YearlyTS {
public:
YearlyTS() : mVec(8760) {}
double& operator[](std::size_t idx) { return mVec[idx];}
// etc.
// Do *NOT* expose resize, clear, push_back, etc.
private:
  std::vector<double> mVec;
};

struct newStructure
{
    YearlyTS foDuration;
    YearlyTS poDuration;
    YearlyTS foRate;
    YearlyTS poRate;
    YearlyTS npoMin;
    YearlyTS npoMax;

    unsigned unitCount;
    float nominalCapacity;

    double forcedVolatility;
    double plannedVolatility;

    Data::ThermalLaw forcedLaw = Data::thermalLawUniform;
    Data::ThermalLaw plannedLaw = Data::thermalLawUniform;
};
```

So you get both the "small object on the stack" and "fixed size" aspects ? Note that on my machine, `sizeof(newStructure) = 176`.